chore(core): allow advanced scss import

madflow opened the pull request:
### Describe the PR

This PR allows importing the sass index file via

```
@import '~bootstrap';
@import '~bootstrap-vue';
```
This is aligned with the `sass-loader` documentation:

https://github.com/webpack-contrib/sass-loader#resolving-import-at-rules

and how Bootstrap enables this feature:

https://github.com/twbs/bootstrap/blob/60559d44a2166708a4dc2f6ccb835052da08ab65/package.json#L74

Hardcoding `node_modules` is not necessary, since Webpack is needed here anyway and the `sass-loader` takes care of the module resolution.

### PR checklist

<!-- (Update "[ ]" to "[x]" to check a box) -->

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement
- [ ] ARIA accessibility
- [x] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [x] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] Includes any needed TypeScript declaration file updates
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [x] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

tmorehouse commented:
@madflow have you tested with other module bundlers?  Some people use Parcel or other bundlers

madflow commented:
@tmorehouse 

I only mentioned Webpack because the docs currently state:

> Note: Requires webpack configuration to load CSS/SCSS files (official guide).

I have done a quick test with Parcel now: It does not seem to work like Webpacks `sass-loader` does and doe not resolve the `sass` field in `package.json`. 

But both Bundlers understand:

```
@import '~bootstrap/scss/bootstrap';
@import '~bootstrap-vue/src/index.scss';
```

The Rollup plugin land is too confusing for me. I only found https://github.com/thgh/rollup-plugin-scss and they chow how to import a `css``file. I would asume that Rollup does not known what "~bootstrap" is.

Maybe amend the docs and clearly state that Bundlers can behave differently and that the above is Webpack + sass-loader use case?

tmorehouse commented:
Yeah, maybe a note about referring to the respective bundler's import name resolution?

madflow commented:
O.K. I changed the docs and rebased - left the initial part basically as it was, and added examples for Webpack and  Parcel. Rollup remains dark magic. I did not go into detail how module paths are resolved because this would add unnecessary complexity (imho) - and in Webpack's case is really complex...

tmorehouse commented on docs/markdown/intro/README.md:
```diff
@@ -156,8 +159,28 @@ Place all of the SCSS `@import`s into a **single SCSS file**, and import that si
+For more details how to configure asset loading and how modules are resolved, please consult the module bundlers documentation.
+
+**Notes**: 
+* Webpack configuration to load CSS files
```
Please use `-` for lists

tmorehouse commented on docs/markdown/intro/README.md:
````diff
@@ -156,8 +159,28 @@ Place all of the SCSS `@import`s into a **single SCSS file**, and import that si
+@import '~bootstrap-vue/src/index.scss';
+```
+
+For more details how to configure asset loading and how modules are resolved, please consult the module bundlers documentation.
````
Lines should be wrapped at 100 characters

tmorehouse commented on docs/markdown/intro/README.md:
```diff
@@ -104,10 +104,11 @@ browsers and devices while providing slightly more opinionated resets to common
-[rollup.js](https://rollupjs.org/), etc, you may prefer to directly include the package into your
-project. To get started, use `yarn` or `npm` to get the latest version of Vue.js, BootstrapVue and
-Bootstrap v4:
+If you are using module bundlers like [Webpack](https://webpack.js.org/),
```
Please wrap lines at 100 characters

tmorehouse commented on docs/markdown/intro/README.md:
```diff
@@ -156,8 +159,28 @@ Place all of the SCSS `@import`s into a **single SCSS file**, and import that si
 functions between files by default.
 
-**Note**: Requires webpack configuration to load CSS/SCSS files
+Webpack and Parcel support prepending the `scss` modules with tilde paths (`~`) when importing from a `scss` file:
```
Lines need to be wrapped at 100 characters

madflow commented:
Updated Prettier, ran Prettier, rebased.
